docs/readmes/resources/ref_magma_metrics.md

```diff
@@ -0,0 +1,186 @@
+[`MetricsExporter`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/cloud/go/services/metricsd/protos/exporter.proto)
+servicers in its `Collect()` method.
+
+```go
+metricsExporters, err := metricsd.GetMetricsExporters()
+...
+for _, e := range metricsExporters {
+	err := e.Submit(metricsToSubmit)
+    ...
+}
+```
+
+With default configuration, `GRPCPushExporterServicer` would be the servicer
+registered for `MetricsExporter` which is registered in the
+[`orchestrator`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/cloud/go/services/orchestrator/orchestrator/main.go?L66)
+cloud service. Finally, the `GRPCPushExporterServicer`, through its `Submit`
+method, pushes metrics over gRPC to their home on Prometheus Edge Hub.
+
+```go
+client, err := s.getClient()
+if err != nil {
+    return err
+}
+_, err = client.Collect(context.Background(), &edge_hub.MetricFamilies{Families: families})
+```
+
+The following diagram shows the call flow for the metrics pipeline on the orc8r:
+
+![Orc8r Metrics](assets/orc8r/orc8r_metrics.png)
+
+On a dev environment, the Prometheus Edge Hub is running at
+[localhost:9090](http://localhost:9090).
+
+### Phase 3: NMS and Grafana
+
+The NMS initially reads metric names, descriptions and their corresponding
+PromQL from
+[`LteMetrics.json`](https://sourcegraph.com/github.com/magma/magma/-/blob/nms/app/packages/magmalte/data/LteMetrics.json).
+Then, in [`Explorer.js`](https://sourcegraph.com/github.com/magma/magma/-/blob/nms/app/packages/magmalte/app/views/metrics/Explorer.js), it filters relevant metrics for the network in
+question using the `/networks/{network_id}/prometheus/series` orc8r Swagger
+endpoint.
+
+```javascript
+// filter only those metrics which are relevant to this network
+const metricsMap = {};
+if (metricSeries != null) {
+    metricSeries.forEach((labelSet: prometheus_labelset) => {
+      metricsMap[labelSet['__name__']] = labelSet;
+    });
+}
+```
+
+This endpoint is part of the `metricsd` orc8r service, and uses the
+[`QueryRestrictor`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/cloud/go/services/metricsd/prometheus/restrictor/query_restrictor.go)
+interface to ensure that only metrics with the right labels (network, gateway
+etc.) are returned.
+
+```go
+// RestrictQuery appends a label selector to each metric in a given query so
+// that only metrics with those labels are returned from the query.
+func (q *QueryRestrictor) RestrictQuery(query string) (string, error) {
+	...
+	promQuery, err := promql.ParseExpr(query)
+	...
+	promql.Inspect(promQuery, q.addRestrictorLabels())
+	return promQuery.String(), nil
+}
+
+```
+
+Finally, in the NMS dashboard > Metrics > Explorer UI, each metric gets a
+[Grafana](https://grafana.com/) `<iframe>` which connects to the Grafana Data
+Source API. In turn, the Grafana Data Source API proxies the parametrized metric
+request to the Prometheus Edge Hub and displays the retrieved metrics.
```

Comment:
These dashboards are actually constructed by us as well -- give a short shoutout to that and include something like https://github.com/magma/magma/tree/c6b85b1f1372bfd0ffd39a061d0459cef3ad9aba/nms/app/packages/magmalte/grafana